src/roadveh_cmd.cpp

```diff
@@ -660,13 +642,15 @@ static RoadVehicle *RoadVehFindCloseTo(RoadVehicle *v, int x, int y, Direction d
 
 	if (front->state == RVSB_WORMHOLE) {
 		for (Vehicle *u : VehiclesOnTile(v->tile)) {
-			EnumCheckRoadVehClose(u, &rvf);
+			FindClosestBlockingRoadVeh(u, &rvf);
 		}
 		for (Vehicle *u : VehiclesOnTile(GetOtherTunnelBridgeEnd(v->tile))) {
-			EnumCheckRoadVehClose(u, &rvf);
+			FindClosestBlockingRoadVeh(u, &rvf);
 		}
 	} else {
-		FindVehicleOnPosXY(x, y, &rvf, EnumCheckRoadVehClose);
+		for (Vehicle *u : VehiclesNearTileXY(x, y, 8)) {
```

Comment:
What's the reason that the threshold should be 8 now instead of 6?

Reply:
8 is the maximum in `dist_x` and `dist_y` in `FindClosestBlockingRoadVeh`.

Reply:
Makes sense but we shouldn't hide fixes inside a Codechange PR.

Reply:
I split off the behavior changes to #14114.